mgr/ConsDefs.pm

```diff
@@ -1179,13 +1179,26 @@
     } else {
 	print "Could not find xml libs\n" if (! $param::quiet);
     }
- #Vc check SSE support
- my $cmd = "touch temp_gccflags.c; $CXX -E -dM -o - temp_gccflags.c | grep -q SSE";
- my $VcCPPFLAGS = " -DVC_IMPL=SSE";
- if ($STAR_HOST_SYS =~ 'gcc432$' || system($cmd)) {# No SSE
-   $VcCPPFLAGS = " -DVC_IMPL=Scalar";
-   if (-e "temp_gccflags.c") {`rm temp_gccflags.c`;}
- }
+
+    $TARGET = "";
+    if ($USE_64BITS){
+	$TARGET = "x86_64";
+    } else {
+	$TARGET = "x86";
+    }
+    my $VcDIR = `/cvmfs/star.sdcc.bnl.gov/star-spack/spack/bin/spack find --format '{prefix}' 'vc\@1.4.1' target=$TARGET`;
```

Comment:
This is a very fragile approach for passing dependencies to the build system. Cons should not know or assume anything about Spack. Hard coding the path to Spack is not going to help it either and that's why the CI fails.

A more robust way is to set everything cons needs outside of cons. See, for example, how I did it by using environment variables (a common approach) for the CI build:

https://github.com/star-bnl/star-sw/blob/6a0d42373a26c1ce21b23dc256c8938bf135cf5a/docker/Dockerfile.root5#L92-L93

Reply:
Yes, I understand that. Your example assumes spack can be foudn in $PATH. I know that is true for CI build. Can I assume the production team also have spack in $PATH?

Reply:
To add spack to PATH one can `source star-spack/setup.sh`. `star-spack` is also installed in `/cvmfs/star.sdcc.bnl.gov`